packages/contentful--react-apps-toolkit/src/SDKProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { FC, ReactElement } from 'react';
+import { init, KnownSDK } from '@contentful/app-sdk';
+import { Spinner } from '@contentful/f36-components';
+
+export const SDKContext = React.createContext<{ sdk: KnownSDK | null }>({ sdk: null });
+
+interface SDKProviderProps {
+  loading?: ReactElement;
+}
+
+export const SDKProvider: FC<SDKProviderProps> = (props) => {
+  const [sdk, setSDK] = React.useState<KnownSDK | undefined>();
+
+  React.useEffect(() => {
+    init(setSDK);
+  }, []);
+
+  if (!sdk) {
+    return props.loading ?? <Spinner />;
```

Comment:
Do we really want to block everything while sdk is loading? Maybe at least create escape hatch - when no `loading` prop is provided, we pass `undefined` sdk and let handle loading inside of the app? Wdyt?

Reply:
I personally think I would prefer to have always an SDK returned from the app, otherwise, the user always needs to add a check for the existence of it. I think that was also the initial plan or not? 🤔

cc @andipaetzold

Reply:
ok 👍